Replace corrupted etcd member

### What happened?

I recently noticed that one of our control plane nodes had some etcd issues. So I decided to replace this machine to get rid of the errors.
So I used the KubeOne guide [Monitoring Etcd Ring and Replacing Corrupted Members](https://docs.kubermatic.com/kubeone/v1.5/guides/etcd-corruption/) to replace this node.

I followed along the steps:

- `kubectl drain --ignore-daemonsets --delete-emptydir-data <node-name>`
- `ssh <username>@<ip-address>`
- `sudo kubeadm reset --force`
- `kubectl delete node <node-name>`

And then I recreated the corrupted node with terraform, created the output and tried to install kubernetes with KubeOne
When I hit:

`kubeone apply --manifest kubeone.yaml -t output.tfjson --verbose`

I got the following error message:

```
INFO[08:59:30 CET] Running cluster probes...
E1118 08:59:31.573377   77793 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x15ca220?, 0x2aedac0})
	k8s.io/apimachinery@v0.24.2/pkg/util/runtime/runtime.go:75 +0x84
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x4000706720?})
	k8s.io/apimachinery@v0.24.2/pkg/util/runtime/runtime.go:49 +0x80
panic({0x15ca220, 0x2aedac0})
	runtime/panic.go:884 +0x20c
k8c.io/kubeone/pkg/tasks.safeguardNodeSelectorsAndTolerations(0x40006121e0)
	k8c.io/kubeone/pkg/tasks/probes.go:177 +0x8c
k8c.io/kubeone/pkg/tasks.safeguard(0x40006121e0)
	k8c.io/kubeone/pkg/tasks/probes.go:150 +0x4b4
k8c.io/kubeone/pkg/tasks.(*Task).Run.func1()
	k8c.io/kubeone/pkg/tasks/task.go:60 +0x9c
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x4000887668, 0x6e55fc})
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:220 +0x24
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1c829c8?, 0x40000520c8?}, 0x40008876b8?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:233 +0x50
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x1585401?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:226 +0x40
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x2540be400, 0x3ff6666666666666, 0x0, 0xa, 0x0}, 0x400011d9c0?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:421 +0x58
k8c.io/kubeone/pkg/tasks.(*Task).Run(0x400011d9c0, 0x40006121e0)
	k8c.io/kubeone/pkg/tasks/task.go:55 +0x100
k8c.io/kubeone/pkg/tasks.Tasks.Run({0x40000f0620, 0x4, 0x0?}, 0x400039f958?)
	k8c.io/kubeone/pkg/tasks/tasks.go:40 +0x108
k8c.io/kubeone/pkg/cmd.runApply(0x40006121e0, 0x40001d4cb0)
	k8c.io/kubeone/pkg/cmd/apply.go:160 +0x16c
k8c.io/kubeone/pkg/cmd.applyCmd.func1(0x400027ac80?, {0x18552f4?, 0x4?, 0x4?})
	k8c.io/kubeone/pkg/cmd/apply.go:93 +0xac
github.com/spf13/cobra.(*Command).execute(0x400027ac80, {0x40003b8f80, 0x4, 0x4})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x4dc
github.com/spf13/cobra.(*Command).ExecuteC(0x400027aa00)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x360
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
k8c.io/kubeone/pkg/cmd.Execute()
	k8c.io/kubeone/pkg/cmd/root.go:55 +0xa4
main.main()
	k8c.io/kubeone/main.go:24 +0x1c
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12e5a3c]

goroutine 1 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x4000706720?})
	k8s.io/apimachinery@v0.24.2/pkg/util/runtime/runtime.go:56 +0xec
panic({0x15ca220, 0x2aedac0})
	runtime/panic.go:884 +0x20c
k8c.io/kubeone/pkg/tasks.safeguardNodeSelectorsAndTolerations(0x40006121e0)
	k8c.io/kubeone/pkg/tasks/probes.go:177 +0x8c
k8c.io/kubeone/pkg/tasks.safeguard(0x40006121e0)
	k8c.io/kubeone/pkg/tasks/probes.go:150 +0x4b4
k8c.io/kubeone/pkg/tasks.(*Task).Run.func1()
	k8c.io/kubeone/pkg/tasks/task.go:60 +0x9c
k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1({0x4000887668, 0x6e55fc})
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:220 +0x24
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext({0x1c829c8?, 0x40000520c8?}, 0x40008876b8?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:233 +0x50
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0x1585401?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:226 +0x40
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x2540be400, 0x3ff6666666666666, 0x0, 0xa, 0x0}, 0x400011d9c0?)
	k8s.io/apimachinery@v0.24.2/pkg/util/wait/wait.go:421 +0x58
k8c.io/kubeone/pkg/tasks.(*Task).Run(0x400011d9c0, 0x40006121e0)
	k8c.io/kubeone/pkg/tasks/task.go:55 +0x100
k8c.io/kubeone/pkg/tasks.Tasks.Run({0x40000f0620, 0x4, 0x0?}, 0x400039f958?)
	k8c.io/kubeone/pkg/tasks/tasks.go:40 +0x108
k8c.io/kubeone/pkg/cmd.runApply(0x40006121e0, 0x40001d4cb0)
	k8c.io/kubeone/pkg/cmd/apply.go:160 +0x16c
k8c.io/kubeone/pkg/cmd.applyCmd.func1(0x400027ac80?, {0x18552f4?, 0x4?, 0x4?})
	k8c.io/kubeone/pkg/cmd/apply.go:93 +0xac
github.com/spf13/cobra.(*Command).execute(0x400027ac80, {0x40003b8f80, 0x4, 0x4})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x4dc
github.com/spf13/cobra.(*Command).ExecuteC(0x400027aa00)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x360
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
k8c.io/kubeone/pkg/cmd.Execute()
	k8c.io/kubeone/pkg/cmd/root.go:55 +0xa4
main.main()
	k8c.io/kubeone/main.go:24 +0x1c
```

<!-- Try to provide as much information as possible.
If you're reporting a security issue, please check the guidelines for reporting security issues:
https://github.com/kubermatic/kubeone/blob/main/CONTRIBUTING.md#reporting-a-security-vulnerability -->

### Expected behavior

KubeOne automatically installs the necessary binaries on the new node and joins it to the existing cluster.

<!-- What did you expected to happen? -->

### How to reproduce the issue?

Follow the steps from above and do not have installed `kubelet` on the new node.

<!-- Please provide as much information as possible, so we can reproduce the issue on our own. -->

### What KubeOne version are you using?

<!-- Please provide output of kubeone version here. -->

<details>

```console
$ kubeone version
{
  "kubeone": {
    "major": "1",
    "minor": "5",
    "gitVersion": "1.5.0",
    "gitCommit": "943341700c12aab3c63371a4b19f71d608eec24a",
    "gitTreeState": "",
    "buildDate": "2022-08-30T11:03:01Z",
    "goVersion": "go1.19",
    "compiler": "gc",
    "platform": "darwin/arm64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "54",
    "gitVersion": "v1.54.0",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}
```

</details>

### Provide your KubeOneCluster manifest here (if applicable)

<!-- Providing the KubeOneCluster manifest will help us reproduce the issue.
Please make sure to redact all secrets (e.g. passwords, URLs...)! -->

<details>

```yaml
# paste manifest here
```

</details>

### What cloud provider are you running on?
Hetzner
<!-- AWS, Azure, DigitalOcean, GCP, Hetzner Cloud, Nutanix, OpenStack, Equinix Metal (Packet), VMware vSphere, Other (e.g. baremetal or non-natively supported provider) -->

### What operating system are you running in your cluster?
Ubuntu 20.04
<!-- Ubuntu 20.04/22.04, CentOS 7, Rocky Linux 8, Flatcar Linux, ... -->

### Additional information

After having a Chat in Kubermatic Slack Channel with @kron4eg we found a workaround to fix the issue by installing kubelet to the node, so that KubeOne doesn't get the NPE anymore.
He asked me to open a bug issue so this can be tracked and fixed in upcoming releases.

<!-- Additional information about the bug you're reporting (optional). -->

--- a/pkg/tasks/probes.go
+++ b/pkg/tasks/probes.go
@@ -174,8 +174,13 @@ func safeguardNodeSelectorsAndTolerations(s *state.State) error {
 		return err
 	}
 
+	liveCP := s.LiveCluster.ControlPlane
+	if len(liveCP) == 0 || liveCP[0].Kubelet.Version == nil {
+		return nil
+	}
+
 	// Run safeguard only when upgrading to Kubernetes 1.24.
-	if targetVersion.Minor() == 24 && s.LiveCluster.ControlPlane[0].Kubelet.Version.Minor() == 23 {
+	if targetVersion.Minor() == 24 && liveCP[0].Kubelet.Version.Minor() == 23 {
 		var pods corev1.PodList
 		// List pods in all namespaces
 		if err := s.DynamicClient.List(s.Context, &pods, dynclient.InNamespace("")); err != nil {

